### PR TITLE
Pull out vendor JS code into a separate, cacheable file

### PR DESCRIPTION
**Pull request by aub**
department-of-veterans-affairs/vets.gov-team#928

The JS code we bring in from vendors doesn't change very often. For that reason, it makes sense to pull it out into its own javascript file so the browser can cache it and hold onto the cache for longer than it would the individual entry files that include our code. Doing so also has the benefit of making the entry files much smaller, so that if a user went to multiple entries then they would download much less code.

Annoyingly, this requires us to specify which libraries we want to include in the vendor package, so that if that list changes then we have to update it manually in the WebPack config. But other than that the change is pretty straightforward.

### CODE REVIEW

**Comment by robbiethegeek-usds** [comment]
LGTM 👍 once @crwallace reviews and approves feel free to merge.

File: config/webpack.config.js
Comment (jbalboni) [inline config/webpack.config.js]: We might want to skip lodash here. We currently have babel and webpack plugins that transform it so we're only including the specific functions in use, but this will include all of lodash for every app.

Comment (aub) [inline config/webpack.config.js]: All right, removed lodash from the list.

**Comment by jbalboni** [comment]
@aub One other thing: because it's common to the other app bundles, the vendor bundle is considered the entry bundle by Webpack, which means it contains a manifest with a list of all the other bundles. This is fine in dev, but in production that manifest will have file names with hashes, which means the vendor bundle will change anytime any of our app bundles change, busting the cache.

I think we would need to use another plugin to pull out that manifest and inline it in the html, like described [here](https://medium.com/@okonetchnikov/long-term-caching-of-static-assets-with-webpack-1ecb139adb95#.ypid6zi4l).

**Comment by aub** [comment]
@jbalboni Ugh, ok I was hoping I could get away without doing that. I had built the thing in production a few times and noticed that the hashes didn't change, but I guess I hadn't done it while changing our code in between. On it.

**Comment by aub** [comment]
@jbalboni Ok, I added a couple more things to make it create consistent hashes and it's working now. I did a test locally by making a spurious change to one of the non-vendor files, rebuilt, and then tested that only the appropriate hashes changed.

**Comment by aub** [comment]
@crwallace want to give final approval on this one?

**Review by crwallace — APPROVED** [review]
@aub this looks great!

**Pull request merged**
